src/main/java/cn/nukkit/command/data/CommandData.java

```diff
@@ -1,26 +1,258 @@
-            return new CommandData();
+        /**
+         * Adds all of the permissions in the passed String collection to the list of allowed permissions,
+         * without clearing the current permissions set.
+         * <p/>Note: A player only needs to have <em>one</em> of the permissions in the list to
+         * be able to execute the command.
+         *
+         * @param permissions Collection of permissions to add
+         * @return Builder instance for chaining calls
+         */
+        public Builder addPermissions(@NonNull String... permissions) {
+            this.perms.addAll(Arrays.asList(permissions));
+            return this;
         }
+
+        /**
+         * Clears the current aliases and sets the passed collection of Strings as the new aliases.
+         *
+         * @param aliases Collection of String aliases for this Command
+         * @return Builder instance for chaining calls
+         */
+        public Builder setAliases(@NonNull String... aliases) {
+            this.aliases = Arrays.asList(aliases);
+            return this;
+        }
+
+        /**
+         * Adds the alias to the set of aliases for this Command.
+         *
+         * @param alias Alias to add
+         * @return Builder instance for chaining calls
+         */
+        public Builder addAlias(@NonNull String alias) {
+            this.aliases.add(alias);
+            return this;
+        }
+
+        /**
+         * Adds all the aliases in the passed String collection, without clearing the current aliases.
+         *
+         * @param aliases Collection of String aliases to add to this Command.
+         * @return Builder instance for chaining calls
+         */
+        public Builder addAliases(@NonNull String... aliases) {
+            this.aliases.addAll(Arrays.asList(aliases));
+            return this;
+        }
+
+        /**
+         * Clears the command overloads and sets the passed collection as the overloads for this Command.
+         *
+         * @param paramSet Collection of {@link CommandParameter} arrays to set as the new overloads
+         * @return Builder instance for chaining calls
+         */
+        public Builder setParams(@NonNull CommandParameter[]... paramSet) {
+            this.overloads = Arrays.asList(paramSet);
+            return this;
+        }
+
+        /**
+         * @param parameters
+         * @return
+         * @see #setParams(CommandParameter[]...)
+         */
+        public Builder setParams(@NonNull List<CommandParameter[]> parameters) {
+            this.overloads = parameters;
+            return this;
+        }
+
+        /**
+         * Adds the set of parameters as a Command overload.
+         *
+         * @param paramSet an Array of {@link CommandParameter} objects representing the command parameters for this overload
+         * @return Builder instance for chaining calls
+         */
+        public Builder addParams(@NonNull CommandParameter[] paramSet) {
```

Comment:
might be nice to have this one be varargs as well (`setParams` already is)

Reply:
Done